full-service/src/service/transaction_builder.rs

```diff
@@ -726,6 +770,154 @@ mod tests {
         assert_eq!(proposal.tx.prefix.outputs.len(), 2); // self and change
     }
 
+    // This test is to ensure that we can send a transaction with a total input
+    // value of > u64::MAX
+    #[test_with_logger]
+    fn test_setting_input_txos_overflow_u64(logger: Logger) {
+        let mut rng: StdRng = SeedableRng::from_seed([20u8; 32]);
+
+        let db_test_context = WalletDbTestContext::default();
+        let wallet_db = db_test_context.get_db_instance(logger.clone());
+        let conn = wallet_db.get_conn().unwrap();
+        let known_recipients: Vec<PublicAddress> = Vec::new();
+        let mut ledger_db = get_test_ledger(5, &known_recipients, 12, &mut rng);
+
+        let _sync_thread = SyncThread::start(ledger_db.clone(), wallet_db.clone(), logger.clone());
+
+        let account_key = random_account_with_seed_values(
+            &wallet_db,
+            &mut ledger_db,
+            &vec![u64::MAX, u64::MAX, u64::MAX],
+            &mut rng,
+            &logger,
+        );
+
+        let txos: Vec<Txo> = Txo::list_for_account(
+            &AccountID::from(&account_key).to_string(),
+            None,
+            None,
+            None,
+            None,
+            None,
+            Some(0),
+            &conn,
+        )
+        .unwrap();
+
+        let (recipient, mut builder) =
+            builder_for_random_recipient(&account_key, &ledger_db, &mut rng);
+
+        // This will create a total recipient value of > u64::MAX, which should be valid
+        builder
+            .add_recipient(recipient.clone(), u64::MAX, Mob::ID)
+            .unwrap();
+
+        builder
+            .add_recipient(recipient.clone(), u64::MAX, Mob::ID)
+            .unwrap();
+
+        builder.set_tombstone(22).unwrap();
+        builder.set_fee(Mob::MINIMUM_FEE, Mob::ID).unwrap();
+
+        // This will create a total input value of > u64::MAX, which should be valid.
+        builder
+            .set_txos(
+                &conn,
+                &vec![txos[0].id.clone(), txos[1].id.clone(), txos[2].id.clone()],
+            )
+            .unwrap();
+
+        // NOTE: We have to use an Empty memo here because the RTH memo will fail
+        // because of a u64::MAX limit on the total_outlay value.
+        // See https://github.com/mobilecoinfoundation/mobilecoin/blob/437133a545b85958278efcb655bce36929c8f72a/transaction/extra/src/memo/destination_with_payment_request_id.rs#L51
+        // for more details.
+        let _unsigned_tx_proposal = builder.build(TransactionMemo::Empty, &conn).unwrap();
```

Comment:
Should validate the change and outputs are correct

Reply:
Why was this marked as resolved?

Reply:
I've updated that to check the change and outputs